add a prediction script

paulfitz opened the pull request:
Great model!

This adds a `predict.py` script to do inference with the model, predicting sql from questions without attempting to evaluate accuracy (which would require that the questions be labeled).

The `train.py` script is tweaked slightly so that the methods within it can be used by `predict.py`.

The `annotate_ws.py` script is also tweaked slightly so that it can be run on user data.

Two utilities, `add_csv.py` and `add_question.py` are added to give a quick way to set up some new tables and questions.

To minimize changes to existing code, I add unlabeled questions with a dummy label, as empty as I could make it while still having working code.

whwang299 commented on predict.py:
```diff
@@ -0,0 +1,138 @@
+path_model_bert = args.bert_model_file
+path_model = args.model_file
+model, model_bert, tokenizer, bert_config = get_models(args, BERT_PT_PATH, trained=True, path_model_bert=path_model_bert, path_model=path_model)
+opt, opt_bert = get_opt(model, model_bert, args)
```
This line can be omitted?

paulfitz replied:
Oh good point, thank you!  Removed.

whwang299 commented on train.py:
```diff
@@ -127,8 +127,8 @@ def get_bert(BERT_PT_PATH, bert_type, do_lower_case, no_pretraining):
 
-def get_opt(model, model_bert, fine_tune):
-    if fine_tune:
+def get_opt(model, model_bert, args):
```
If the line indicated above in `predict.py` can be omitted, these change are not necessary I guess.

whwang299 commented:
Hi @paulfitz 

Thanks for the contribution! Before merge, I found some lines of code that may be omitted. Please let me know your opinion. 

Thanks!

paulfitz commented:
> Thanks for the contribution! Before merge, I found some lines of code that may be omitted. Please let me know your opinion.

Thanks for reviewing!  I've removed get_opt from `predict.py` and reverted `train.py` - thanks for spotting that, this is cleaner.
